### PR TITLE
Fixed crash in debug info

### PR DESCRIPTION
**Pull request by homuroll**

### CODE REVIEW

File: llvmDebugInfoC/src/main/cpp/DebugInfoC.cpp [hunk 1]
```diff
@@ -152,8 +146,10 @@ DICompositeTypeRef DICreateArrayType(DIBuilderRef refBuilder,
                                      uint64_t elementsCount) {
   auto builder = llvm::unwrap(refBuilder);
   auto range = std::vector<llvm::Metadata*>({llvm::dyn_cast<llvm::Metadata>(builder->getOrCreateSubrange(0, size))});
-  return llvm::wrap(builder->createArrayType(size, alignInBits, llvm::unwrap(refType),
-                                             builder->getOrCreateArray(range)));
+  auto type = builder->createArrayType(size, alignInBits, llvm::unwrap(refType),
+                                                           builder->getOrCreateArray(range));
+  builder->retainType(type);
```
Comment (SvyatoslavScherbina) [inline llvmDebugInfoC/src/main/cpp/DebugInfoC.cpp]: Is the same required for `DICreateMemberType`?

File: llvmDebugInfoC/src/main/cpp/DebugInfoC.cpp [hunk 2]
```diff
@@ -128,21 +128,15 @@ DICompositeTypeRef DICreateStructType(DIBuilderRef refBuilder,
...
-  auto elementsArray = builder->getOrCreateArray(typeElements);
-  auto composite = builder->createStructType(llvm::unwrap(scope),
-                                              name, llvm::unwrap(file),
-                                              lineNumber,
-                                              sizeInBits, alignInBits, (llvm::DINode::DIFlags)flags,
-                                              llvm::unwrap(derivedFrom),
-                                              elementsArray);
-  builder->replaceTemporary(llvm::TempDIType(llvm::unwrap(refPlace)), composite);
-  return llvm::wrap(composite);
+  *(int*)1=1;
```
Comment (SvyatoslavScherbina) [inline llvmDebugInfoC/src/main/cpp/DebugInfoC.cpp]: ?

File: llvmDebugInfoC/src/main/cpp/DebugInfoC.cpp (hunk 1 above)
Comment (homuroll) [reply llvmDebugInfoC/src/main/cpp/DebugInfoC.cpp]: No.

File: llvmDebugInfoC/src/main/cpp/DebugInfoC.cpp [hunk 3]
```diff
@@ -53,7 +54,7 @@ DEFINE_SIMPLE_CONVERSION_FUNCTIONS(DIExpression,     DIExpressionRef)
 extern "C" {
 
 DIBuilderRef DICreateBuilder(LLVMModuleRef module) {
-  return llvm::wrap(new llvm::DIBuilder(* llvm::unwrap(module)));
+  return LLVMCreateDIBuilder(module);
```
Comment (olonho) [inline llvmDebugInfoC/src/main/cpp/DebugInfoC.cpp]: Maybe call `LLVMCreateDIBuilder` directly from users of `DICreateBuilder`?

File: llvmDebugInfoC/src/main/cpp/DebugInfoC.cpp [hunk 4]
```diff
@@ -62,8 +63,7 @@ void DIFinalize(DIBuilderRef builder) {
 }
 
 void DIDispose(DIBuilderRef builder) {
-  auto diBuilder = llvm::unwrap(builder);
-  delete diBuilder;
+  LLVMDisposeDIBuilder(builder);
```
Comment (olonho) [inline llvmDebugInfoC/src/main/cpp/DebugInfoC.cpp]: Same as above.

**Pull request merged**